### PR TITLE
add helper functions to UrlUtil

### PR DESCRIPTION
**Pull request by vitriolix**
this adds a few helper functions I'm using in GFDroid that might be helpful to others too.

### CODE REVIEW

File: android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt [hunk 1]
```diff
@@ -149,5 +149,63 @@ class UrlUtil {
...
+                return split
+            }
+            return null
+        }
+
+        @JvmStatic
+        fun getSanitizedUrlList(urls: List<String>): ArrayList<String> {
+            val sanitizedUrls = ArrayList<String>()
+            for (url in urls)
+                sanitizedUrls += urls.joinToString(separator = ",", transform = { sanitizeUrl(it).take(30) })
```
Comment (mnbogner) [inline android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt:167]: i don't think that the truncation is necessary now that the urls are being shortened by being sanitized.  the longest output i could find came to 34 characters.

File: android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt [hunk 2]
```diff
@@ -149,5 +149,63 @@ class UrlUtil {
 
             return sanitizedString
         }
+
+        @JvmStatic
+        fun getServiceFromUrl(url: String): String? {
```
Comment (mnbogner) [inline android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt:154]: where is this method used?

File: android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt [hunk 3]
```diff
@@ -149,5 +149,63 @@ class UrlUtil {
...
+
+        @JvmStatic
+        fun joinToTruncatedString(list: List<String>, length: Int): String {
+            return list.joinToString(separator = ",", transform = { truncate(it, length) })
+        }
+
+        @JvmStatic
+        fun joinToString(urls: ArrayList<String>): String {
+            return urls.joinToString(",")
+        }
```
Comment (mnbogner) [inline android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt:209]: i'm not sure if i agree that these methods that are pure string manipulation belong in envoy,

File: android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt (hunk 2 above)
Comment (vitriolix) [reply android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt:154]: it's used in a few places in gf-droid code as well as in joinToSanitizedAndTruncatedString and sanitizeUrl which is used a few places

File: android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt (hunk 3 above)
Comment (vitriolix) [reply android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt]: good point, will move joinToString and joinToTruncatedString into my generic utils class, the others seem appropriate though as they depend on other UrlUtils methods like `sanitize()`

File: android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt (hunk 1 above)
Comment (vitriolix) [reply android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt]: ok, removed

**Review by mnbogner — APPROVED** [review]
:ship:

**Pull request merged**